tool/expose_version.dart

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env dart
+import 'dart:io';
+
+void main(List<String> packages) {
+  for (var package in packages) {
+    final libraryFile = File('./$package/lib/$package.dart');
+    final content = libraryFile.readAsStringSync();
+    final strippedContent = stripVersion(content);
+
+    final pubspecFile = File('./$package/pubspec.yaml');
+    final projectData = pubspecFile.readAsStringSync();
+    final versionNumber = resolveVersion(projectData);
+    final versionVariable = createVersionVariable(versionNumber);
+
+    libraryFile.writeAsStringSync('$strippedContent\n$versionVariable\n');
+  }
+}
+
+String createVersionVariable(String versionNumber) =>
+    "const version = '$versionNumber';";
+
+String resolveVersion(String projectData) =>
+    RegExp('version: ([0-9]*\.[0-9]*\.[0-9]*)')
+        .firstMatch(projectData)
+        .group(1);
+
+String stripVersion(String content) => content.replaceAll(
```

Comment:
The regex should also include the pre release version strings

Reply:
I removed the complexity of verifying the smever all together.